sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -250,174 +241,70 @@ export class EventHubReceiver extends LinkEntity {
-          "Hence not calling detached from the _onAmqpClose() handler.",
-        this._context.connectionId,
-        this.name,
-        this.address
-      );
+    logger.verbose(
+      "[%s] 'receiver_close' event occurred on the receiver '%s' with address '%s'. " +
+        "Value for isItselfClosed on the receiver is: '%s'.",
+      this._context.connectionId,
+      this.name,
+      this.address,
+      rheaReceiver ? rheaReceiver.isItselfClosed().toString() : undefined
+    );
+    if (rheaReceiver && !this.isConnecting) {
+      // Call close to clean up timers & other resources
+      await rheaReceiver.close();
     }
   }
 
   private async _onAmqpSessionClose(context: EventContext): Promise<void> {
     const rheaReceiver = this._receiver || context.receiver;
-    if (!rheaReceiver || rheaReceiver.isSessionItselfClosed()) {
-      logger.verbose(
-        "[%s] 'session_close' event occurred on the session of receiver '%s' with " +
-          "address '%s' and the sdk did not initiate this. Moreover the receiver is already " +
-          "re-connecting. Hence not calling detached from the _onAmqpSessionClose() handler.",
-        this._context.connectionId,
-        this.name,
-        this.address
-      );
-      return;
-    }
-
-    const sessionError = context.session && context.session.error;
-    if (sessionError) {
-      logger.verbose(
-        "[%s] 'session_close' event occurred for receiver '%s' with address '%s'. " +
-          "The associated error is: %O",
-        this._context.connectionId,
-        this.name,
-        this.address,
-        sessionError
-      );
-    }
-
-    if (!this.isConnecting) {
-      logger.verbose(
-        "[%s] 'session_close' event occurred on the session of receiver '%s' with " +
-          "address '%s' and the sdk did not initiate this. Hence calling detached from the " +
-          "_onAmqpSessionClose() handler.",
-        this._context.connectionId,
-        this.name,
-        this.address
-      );
-      await this.onDetached(sessionError);
-    } else {
-      logger.verbose(
-        "[%s] 'session_close' event occurred on the session of receiver '%s' with " +
-          "address '%s' and the sdk did not initiate this. Moreover the receiver is already " +
-          "re-connecting. Hence not calling detached from the _onAmqpSessionClose() handler.",
-        this._context.connectionId,
-        this.name,
-        this.address
-      );
+    logger.verbose(
+      "[%s] 'session_close' event occurred on the session of receiver '%s' with address '%s'. " +
+        "Value for isSessionItselfClosed on the session is: '%s'",
+      this._context.connectionId,
+      this.name,
+      this.address,
+      rheaReceiver ? rheaReceiver.isSessionItselfClosed().toString() : undefined
+    );
+    if (rheaReceiver && !this.isConnecting) {
+      // Call close to clean up timers & other resources
+      await rheaReceiver.close();
```

Comment:
Same comment here about how to handle errors from close.

Reply:
Fixed with https://github.com/Azure/azure-sdk-for-js/pull/6737/commits/06e9251b0941110ccdaa398a637d4fba818f69f1